hbase-client/src/main/java/org/apache/hadoop/hbase/client/CatalogReplicaLoadBalanceReplicaSelector.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.client;
+
+import org.apache.hadoop.hbase.HRegionLocation;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * A Catalog replica selector decides which catalog replica to go for read requests.
+ */
+@InterfaceAudience.Private
+public interface CatalogReplicaLoadBalanceReplicaSelector {
+
+  /**
+   *  This method is called upon when input location is stale.
+   * @param loc location which is stale.
+   * @param fromReplicaId which replica the stale location comes from.
+   */
+  void notifyOnError(HRegionLocation loc, int fromReplicaId);
```

Comment:
Do we need the exception?

Reply:
Yeah, do we need to pass in the exception?

Reply:
s/notifyOnError/onError/ ?

Reply:
Hmm, I do not think we need the Exception here. The caller already filters out RegionMovedException. Is there anything missing?

Reply:
NotServingRegionException? Anyway, this is an IA.Private class so it can be changed at any time. I think we could move the filtering logic down to the implementation class.

Reply:
My thoughts was that since filtering logic is done already, it is not necessary to do for a second time in selector implementation. Some of the meta data such as "fromMetaReplicaId" needs to come from TableCache (in the future implementation), so somehow, upper layer is already aware of RegionMovedException and NotServingRegionException. 
Will keep it as it is for now.